applications/serial_lte_modem/src/main.c

```diff
@@ -124,30 +123,85 @@ static int ext_xtal_control(bool xtal_on)
 	return err;
 }
 
-/* This function is safe to call twice after reset */
-static int init_gpio(void)
+#if CONFIG_SLM_POWER_PIN >= 0
+static void power_pin_callback(const struct device *, struct gpio_callback *, uint32_t)
+{
+	LOG_INF("Power pin triggered.");
```

Comment:
It seems logical that the interrupt gets called repeatedly if it is configured with a level and not an edge trigger.

Reply:
It does not always trigger multiple times.

However, as it seems that you have to change this to level triggered, you have to take into account scenarios where the button is not released while the device goes to sleep / wakes up. I think that wakeup is used in some automation and if you combine the pins, then you have to deal with the situation that it is not released in timely manner.

Reply:
Actually the (poweroff) interrupt is configured to trigger on a rising edge, and because the pin is configured with a pull up, the interrupt triggers only when the pin has been toggled both ways. So I did take into account the releasing phase.

Reply:
That is good, however, my comment was about after this has been changed from edge to level triggered.

Reply:
Ah. Right.